REGRESSION(262669@main): Spelling corrected despite "Correct spelling automatically" being turned off

nt1m opened the pull request:
#### 655ad809d943553a160ae45fdb0918b12f4f6868
<pre>
REGRESSION(262669@main): Spelling corrected despite &quot;Correct spelling automatically&quot; being turned off
<a href="https://bugs.webkit.org/show_bug.cgi?id=258305">https://bugs.webkit.org/show_bug.cgi?id=258305</a>
rdar://110217888

Reviewed by Aditya Keerthi.

NSCorrectionIndicatorTypeDefault &amp; NSCorrectionIndicatorTypeReversion bubbles are associated with autocorrection, and have
the accept on dismiss behavior. Do not show those bubbles when autocorrection is turned off.

Only display NSCorrectionIndicatorTypeGuesses bubbles to match behavior from other apps.

* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::AlternativeTextController::startAlternativeTextUITimer):

Canonical link: <a href="https://commits.webkit.org/265330@main">https://commits.webkit.org/265330@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff3af4bf51ee5fcc3894f75b8c53aad996366a6c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10631 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10852 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11135 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12278 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10202 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10646 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13225 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10822 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13123 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10792 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11713 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8940 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12681 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9007 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9598 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16859 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10078 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9748 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13003 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10218 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9371 "Built successfully") | | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2541 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13636 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10074 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/editing/AlternativeTextController.cpp:
```diff
@@ -101,6 +101,10 @@ void AlternativeTextController::startAlternativeTextUITimer(AlternativeTextType
 #if ENABLE(ALTERNATIVE_TEXT_REQUIRES_AUTOMATIC_SPELLING_CORRECTION)
```
Can we move this inside `!isAutomaticSpellingCorrectionEnabled()` to reduce code duplication?

nt1m committed:
REGRESSION(262669@main): Spelling corrected despite "Correct spelling automatically" being turned off
https://bugs.webkit.org/show_bug.cgi?id=258305
rdar://110217888

Reviewed by Aditya Keerthi.

NSCorrectionIndicatorTypeDefault & NSCorrectionIndicatorTypeReversion bubbles are associated with autocorrection, and have
the accept on dismiss behavior. Do not show those bubbles when autocorrection is turned off.

Only display NSCorrectionIndicatorTypeGuesses bubbles to match behavior from other apps.

* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::AlternativeTextController::startAlternativeTextUITimer):

Canonical link: https://commits.webkit.org/265330@main
